### PR TITLE
#434: Make cluster swap relative threshold a configurable parameter

### LINKED ISSUE #434: Make cluster swap relative threshold a configurable parameter
As suggested by @lifflander in PR #433 , the 5% value in line 117 0f [Execution/lbsClusteringTransferStrategy.py](https://github.com/DARMA-tasking/LB-analysis-framework/pull/433/files/41e8215c721da7c1c729dca393b8affe55f75675#diff-52bdf682017c84ecfdd2f4caa52bf087545beaee905a7211a4ec57b9731a934c) should be made an optional (with default value of 0.05) user-defined parameter

A suggested name is `cluster_swap_rtol`. 

Please make sure that CI is updated accordingly and that this new configuration parameter is tested for invalid types.

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #434

### CODE REVIEW

File: src/lbaf/Execution/lbsClusteringTransferStrategy.py [hunk 1]
```diff
@@ -40,8 +43,9 @@ def __cluster_objects(self, rank):
             k: clusters[k]
             for k in random.sample(clusters.keys(), len(clusters))}
 
-    def __find_suitable_subclusters(self, clusters, rank_load, r_tol=0.05):
```
Comment (cwschilly) [inline src/lbaf/Execution/lbsClusteringTransferStrategy.py:43]: Will it be a problem that I removed this parameter? From what I could tell, it is never explicitly set (ie only the default value is used). I could of course revert back to how it was, or even set it to the new `cluster_swap_rtol`

File: src/lbaf/Execution/lbsTransferStrategyBase.py [hunk 2]
```diff
@@ -147,7 +148,10 @@ def factory(
         try:
             # Instantiate and return object
             strategy = locals()[strategy_name + "TransferStrategy"]
-            return strategy(criterion, parameters, logger)
+            if strategy_name == "Clustering":
```
Comment (cwschilly) [inline src/lbaf/Execution/lbsTransferStrategyBase.py:151]: Is there a more elegant solution than this `if` block?

File: src/lbaf/Execution/lbsClusteringTransferStrategy.py (hunk 1 above)
Comment (ppebay) [reply src/lbaf/Execution/lbsClusteringTransferStrategy.py:43]: `r_tol` is the value that should be passed by the configuration parameter. Whether you want to pass it, once retrieved, to this function as is currently done; or rather store it as member variable of the class, is an implementation detail I will let you decide based on what you think is most convenient.

File: src/lbaf/Execution/lbsTransferStrategyBase.py (hunk 2 above)
Comment (ppebay) [reply src/lbaf/Execution/lbsTransferStrategyBase.py:151]: In the `develop` branch I still seem to have the following line 150:
```
 return strategy(criterion, parameters, logger)
```
at commit `0cd1502c45e683eda9963896251c7d9c87d330ac`

File: src/lbaf/Execution/lbsClusteringTransferStrategy.py (hunk 1 above)
Comment (cwschilly) [reply src/lbaf/Execution/lbsClusteringTransferStrategy.py:43]: Resolved

**Review by ppebay — COMMENTED** [review]
Is this ready for final review (it's still in draft mode)

**Comment by cwschilly** [comment]
@ppebay This PR is now ready for review

**Review by ppebay — APPROVED** [review]
Additions to CI look good, thanks @cwschilly

**Pull request merged**